constraints/exhaustivity.js

```diff
@@ -23,8 +23,34 @@ function exhaust1(s, ptree){
 			if (ptree.cat!==child.cat && pCat.nextLower(ptree.cat)!==child.cat){
 				vcount++;
 			}
-			vcount += exhaust1(s, child);
+			vcount += exhaustChild(s, child);
 		}
 		return vcount;
 	}
-};
\ No newline at end of file
+};
+
+function exhaustParent(s, ptree){
+//Assumes trees that obey Layering.
+	//Base case: if parent is a terminal, return 0 violations.
+	if (!ptree.children){
+		return 0;
+	}
+	
+	//Recursive case: if parent is non_terminal, find out if there are any violations in each of the subtrees rooted in its children.
+
+	if(ptree.children && ptree.children.length){
+		var vcount = 0;
+		var child;
+		alreadyViolated = [];
+		for (var i=0; i < ptree.children.length; i++){
+			
+			child = ptree.children[i];
+			if (ptree.cat!==child.cat && pCat.nextLower(ptree.cat)!==child.cat && !alreadyViolated.includes(child.cat)){
```

Comment:
This isn't quite right either, because there could be, for example, multiple phis that contain clitics. You could change this to push the id of the parent instead of the category, but I think it would be more efficient to use a while loop with a boolean flag. Take a look at the implementation of eqSis and strongStart in equalSisters.js and strongStart.js respectively.